tgtg/__init__.py

```diff
@@ -96,57 +100,97 @@ def _refresh_token(self):
 
     def login(self):
         if not (
-            self.password
-            and self.email
-            or self.access_token
-            and self.refresh_token
-            and self.user_id
+                self.email
+                or self.access_token
+                and self.refresh_token
+                and self.user_id
         ):
             raise TypeError(
-                "You must provide at least email and password or access_token, refresh_token and user_id"
+                "You must provide at least email or access_token, refresh_token and user_id"
             )
 
         if self._already_logged:
             self._refresh_token()
         else:
             response = requests.post(
-                self._get_url(LOGIN_ENDPOINT),
+                self._get_url(AUTH_BY_EMAIL_ENDPOINT),
                 headers=self._headers,
                 json={
                     "device_type": "ANDROID",
                     "email": self.email,
-                    "password": self.password,
                 },
                 proxies=self.proxies,
                 timeout=self.timeout,
             )
             if response.status_code == HTTPStatus.OK:
-                login_response = response.json()
-                self.access_token = login_response["access_token"]
-                self.refresh_token = login_response["refresh_token"]
-                self.last_time_token_refreshed = datetime.datetime.now()
-                self.user_id = login_response["startup_data"]["user"]["user_id"]
+                first_login_response = response.json()
+
+                if first_login_response["state"] == "TERMS":
+                    raise TgtgPollingError("Please accept terms first, validate your email and then retry!")
+
+                self.start_polling(first_login_response["polling_id"])
+
             else:
-                raise TgtgLoginError(response.status_code, response.content)
+                if response.status_code == 429:
+                    raise TgtgAPIError("429 - Too many requests. Try again later.")
+                else:
+                    raise TgtgLoginError(response.status_code, response.content)
+
+    def start_polling(self, polling_id):
+        polling_response_code = 202
+        i = 1
+
+        while polling_response_code == 202 and i <= MAX_POLLING_TRIES:
```

Comment:
can you please use a forloop here ?
something like:
```
for retry_index in range(MAX_POLLING_TRIES):
```

Reply:
Fixed!